File: POS-API/POS.Domain/Entities/Customer.cs
using POS.Domain.Common;

namespace POS.Domain.Entities
{
    public class Customer : BaseEntity
    {
        public Guid CompanyId { get; set; }
        public Guid TenantId { get; set; } // Added for consistency
        public string CustomerCode { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string? LastName { get; set; }
        public string? CompanyName { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? AddressLine1 { get; set; }
        public string? City { get; set; }
        public bool IsActive { get; set; } = true;
        public decimal LoyaltyPoints { get; set; }
        public Guid? CreatedBy { get; set; }

        public Company? Company { get; set; }
    }
}

File: pos-web/src/types/customer.ts
export interface Customer {
    id: string;
    companyId: string;
    customerCode: string;
    firstName: string;
    lastName?: string;
    companyName?: string;
    email?: string;
    phone?: string;
    addressLine1?: string;
    city?: string;
    loyaltyPoints: number;
    isActive: boolean;
    createdAt: string;
}

export interface CreateCustomerDto {
    firstName: string;
    lastName?: string;
    companyName?: string;
    email?: string;
    phone?: string;
    addressLine1?: string;
    city?: string;
    customerCode: string;
}
